ts/private/ts_config.bzl

```diff
@@ -50,9 +50,9 @@ def _ts_config_impl(ctx):
         if TsConfigInfo in dep:
             transitive_deps.append(dep[TsConfigInfo].deps)
 
-    transitive_sources = js_lib_helpers.gather_transitive_sources([], ctx.attr.deps)
+    transitive_sources = js_lib_helpers.gather_transitive_sources(files, ctx.attr.deps)
```

Comment:
I would even consider dropping this completely and only exposing things in `JsInfo` that would be needed for compilation. The tsconfig may be needed in composite projects, yeah, but for that you have the whole concept of the `TsConfigInfo` provider?

Reply:
A `ts_config` target needs to expose a tsconfig file and all it's dependencies (more tsconfig files/targets such as `extends` or `composite`, sometimes npm packages referenced by things like `compilerOptions.types` etc).

Today that is awkwardly duplicated in `TsConfigInfo` and `JsInfo` with a TODO to potentially drop the extra `TsConfigInfo` (#421). IMO we should drop one or the other from here and remove the confusing duplication 🤷